.github/workflows/integration-tests.yml

```diff
@@ -13,22 +13,25 @@ jobs:
         uses: actions/checkout@v4
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v3
-      - name: Build images
+      # Read cache on PRs, write cache on main. Keeps our cache smaller and more stable.
+      - name: Build and cache images
```

Comment:
We should not build them here at all, instead, we must pull the images that are built on previous stages of the workflow. This will require joining everything into a single high-level workflow to orchestrate things.

Reply:
Yep, check latest push 😉